### PR TITLE
#774 update import_file script to add provider

### PR DESCRIPTION
**Pull request by vancamti**

### CODE REVIEW

**Review by koenedaele — APPROVED** [review]
Looks good to me.

**Review by Wim-De-Clercq — CHANGES_REQUESTED** [review]
I'd say that in general, I would avoid underscores on commandline. It's more common to use hyphens.
However, considering the use of underscores is already prevalent in previously existing commands, I guess I'll keep an eye closed for this :see_no_evil: 

----

The more common way of doing boolean commandline parameters is to have a 
parameter per True/False value. For example `--create-provider` and `--no-create-provider` without a value.

If this was a >=3.9 package this could be done in 1 line:
```py
parser.add_argument('--create-provider', action=argparse.BooleanOptionalAction)
```

As atramhasis appears to be >=3.8 this is done as:
```py
parser.add_argument('--create-provider', action='store_true')
# I'm not a 100% sure whether the `dest` is with underscore or hyphen
parser.add_argument('--no-create-provider', dest='create-provider', action='store_false')
parser.set_defaults(create_provider=True)  # can maybe be set on line 1.
```

**Comment by koenedaele** [comment]
I'm fine with making 2.0.0 compatible with python >= 3.9. See #840

**Comment by vancamti** [comment]
> I'd say that in general, I would avoid underscores on commandline. It's more common to use hyphens.

I changed the existing arguments and new arguments to contain - instead of _. I prefer do it right unless it breaks something insurmountable.

**Review by Wim-De-Clercq — CHANGES_REQUESTED** [review]
Yea, alright. I suppose because we have a major release jump it's acceptable to change existing parameters.

While we're at it. I would also make the required "options" more clear.
Currently, there is no way to see that `--from` and `--uri-pattern` are required from the help output.

```
options:
  -h, --help            show this help message and exit
  --from INPUT_FILE     local path to the input file
  ...
  --uri-pattern URI_PATTERN
```
And even worse is that in 3.9 the subtitle "options" is called "optional arguments".

This is because, rightfully so, the library assumes anything starting with `--` is an option / optional. Parameters which are required are normally positionals without the `--` prefix.

The solution is quite simple, either we make from and uri positionals (just remove the `--` from the name), or we can add a special group with required options:
```py
    required_args = parser.add_argument_group('required named arguments')
    required_args.add_argument(
        '--from',
        dest='input_file',
        type=str,
        help='local path to the input file',
        required=True
    )
```
Then it will place those parameters under their own section:
```
required named arguments:
  --from INPUT_FILE     local path to the input file
  --uri-pattern URI_PATTERN
```

I'd argue for the positionals, because it's the standard way of doing things.

----

`docs/source/customisation.rst` also contains text about this script which should get updates I suppose.

File: atramhasis/scripts/import_file.py [hunk 1]
```diff
@@ -106,48 +106,77 @@ def parse_argv_for_import(argv):
         description='Import file to a database',
         epilog=(
             f'example: {cmd} '
-            '--from atramhasis/scripts/my_file '
+            'atramhasis/scripts/my_file '
+            'urn:x-skosprovider:trees:%s'
```
Comment (Wim-De-Clercq) [inline atramhasis/scripts/import_file.py:110]: \+ space at the end of line 110 string.

**Review by Wim-De-Clercq — CHANGES_REQUESTED** [review]
Just something minor

**Review by Wim-De-Clercq — APPROVED** [review]

**Review by koenedaele — CHANGES_REQUESTED** [review]
I imported a CS:
```
import_file --from fixtures/onderwijsstructuur-final.skos.ttl --to sqlite:///atramhasis.sqlite --uri_pattern https://w3id.org/onderwijs-vlaanderen/id/structuur/ --create_provider True --provider_id OSTR --id_generation_strategy manual
```

This worked, but runs fine, but there's nothing in the DB afterwards (no conceptscheme or provider).

At the end I do get the message:

```
** The import of conceptscheme ONDERWIJSSTRUCTUUR-FINAL.SKOS from the fixtures/onderwijsstructuur-final.skos.ttl file to sqlite:///atramhasis.sqlite was succesful. ***          

To use the data in Atramhasis, you must edit the file my_thesaurus/skos/__init__.py.          
Add a configuration similar to:            

def create_registry(request):            
	# create the SKOS registry            
	registry = Registry(instance_scope='threaded_thread')            
	ONDERWIJSSTRUCTUUR-FINAL.SKOS = SQLAlchemyProvider(            
		{'id': 'ONDERWIJSSTRUCTUUR-FINAL.SKOS', 'conceptscheme_id': 12},            
		request.db            
	)            
	registry.register_provider(ONDERWIJSSTRUCTUUR-FINAL.SKOS)            
	return registry
```

This is clearly wrong. If create_provider, then they should continue through the UI. If not create_provider, they should  fix things in the database.

**Comment by koenedaele** [comment]
Sorry, I forgot to pull the PR first. Now the command is:

```
import_file fixtures/onderwijsstructuur-final.skos.ttl https://w3id.org/onderwijs-vlaanderen/id/structuur/ --provider-id OSTR --id-generation-strategy manual --to sqlite:///atramhasis.sqlite --create-provider
```

Still no conceptscheme or provider in the DB afterwards though.

**Comment by koenedaele** [comment]
> I assume we are missing a `session.commit()`. When no provider is created it should print the long message
> 
> ```
> print(
>             "\n\n*** The import of conceptscheme {0} from the {1} file to {2} was succesful. ***\
>               \n\nTo use the data in Atramhasis, you must edit the file my_thesaurus/skos/__init__.py.\
>               \nAdd a configuration similar to:\
>                 \n\ndef create_registry(request):\
>                 \n\t# create the SKOS registry\
>                 \n\tregistry = Registry(instance_scope='threaded_thread')\
>                 \n\t{3} = SQLAlchemyProvider(\
>                 \n\t\t{{'id': '{4}', 'conceptscheme_id': {5}}},\
>                 \n\t\trequest.db\
>                 \n\t)\
>                 \n\tregistry.register_provider({6})\
>                 \n\treturn registry\
>                 \n\n".
>             format(
>                 prov_id, args.input_file, args.to,
>                 prov_id.replace(' ', '_'), prov_id, scheme_id, prov_id.replace(' ', '_')
>             )
>         )
> ```
> 
> otherwise we can just print `*** The import of conceptscheme {0} from the {1} file to {2} was succesful. *** \n You can now continue through the Atramhasis UI`?

Agreed. But the ID I get in the long description is wrong as well. I should have gotten OSTR, but I got ONDERWIJSSTRUCTUUR-FINAL.SKOS.

**Comment by koenedaele** [comment]
Conceptscheme and Provider do get into the DB now. I do see the metadata.id is not correct:
```
OSTR|12|https://w3id.org/onderwijs-vlaanderen/id/structuur/|{"id": "ONDERWIJSSTRUCTUUR-FINAL.SKOS", "subject": [], "atramhasis.id_generation_strategy": "MANUAL"}|RECURSE
```

So, provider.id is correct, provider.metadata.id is not.

**Comment by vancamti** [comment]
meta_id has been set to `--provider-id'` but when no provider is created and no provider-id it will still default to 
```
input_file = kwargs.get('input_file')
input_name.upper()
```
like it used to be

**Review by koenedaele — APPROVED** [review]

**Comment by koenedaele** [comment]
Ok, works fine now, ready for merge. Thank you!

**Pull request merged**